streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java

```diff
@@ -64,50 +97,98 @@ public static TaskId parse(final String taskIdStr) {
     /**
      * @throws IOException if cannot write to output stream
      */
-    public void writeTo(final DataOutputStream out) throws IOException {
+    public void writeTo(final DataOutputStream out, final int version) throws IOException {
         out.writeInt(topicGroupId);
         out.writeInt(partition);
+        if (version >= MIN_NAMED_TOPOLOGY_VERSION) {
```

Comment:
Although I did strip out the protocol change, I left this in since it doesn't affect anything until we actually bump the protocol version (which I did take out of this PR)